src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ResettableServicePass.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
```

Comment:
I propose to move the pass and the event subscriber in HttpKernel, to make it easier to reuse.

Reply:
Good idea. In this case, tag name and method attribute should be constructor parameters of the pass, I guess.

Reply:
for the tag name yes, that's what we do usually,
for the attribute name, no need IMHO